types/batch.go

```diff
@@ -7,8 +7,45 @@ const (
 // Batch defines a struct for block aggregation for support of batching.
 // TODO: maybe change to BlockBatch
 type Batch struct {
-	StartHeight uint64
-	EndHeight   uint64
-	Blocks      []*Block
-	Commits     []*Commit
+	Blocks  []*Block
+	Commits []*Commit
+}
+
+// StartHeight is the height of the first block in the batch.
+func (b Batch) StartHeight() uint64 {
+	if len(b.Blocks) == 0 {
+		return 0
+	}
+	return b.Blocks[0].Header.Height
+}
+
+// EndHeight is the height of the last block in the batch
+func (b Batch) EndHeight() uint64 {
+	if len(b.Blocks) == 0 {
+		return 0
+	}
+	return b.Blocks[len(b.Blocks)-1].Header.Height
+}
+
+// NumBlocks is the number of blocks in the batch
+func (b Batch) NumBlocks() uint64 {
+	return uint64(len(b.Blocks))
+}
+
+// SizeBytesEstimate returns the sum of the size of bytes of the blocks and commits
+// The actual size of the batch may be different due to additional metadata and protobuf
+// optimizations.
+func (b Batch) SizeBytesEstimate() int {
```

Comment:
I think the name is a bit confusing as it's the size of the inner blocks + commits, which is important to be accurate and not estimate to avoid a every growing tail. 
I'd suggest something like `SizeBlockAndCommitBytes` or `SizeInternalBytes` or somethign which doesn't say `estimate`.